benchmarks/src/jmh/java/software/amazon/jdbc/benchmarks/ReadWriteSplittingLoadBenchmarks.java

```diff
@@ -0,0 +1,184 @@
+package software.amazon.jdbc.benchmarks;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Logger;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Warmup;
+import org.junit.jupiter.api.BeforeAll;
+
+import software.amazon.jdbc.Driver;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import org.postgresql.PGProperty;
+
+import software.amazon.jdbc.PropertyDefinition;
+import software.amazon.jdbc.plugin.readwritesplitting.ReadWriteSplittingPlugin;
+import software.amazon.jdbc.util.StringUtils;
+
+public class ReadWriteSplittingLoadBenchmarks {
+
+  public static final String POSTGRESQL_CONNECTION_STRING =
+      "jdbc:aws-wrapper:postgresql://atlas-postgres.cluster-czygpppufgy4.us-east-2.rds.amazonaws" +
+          ".com:5432/postgres";
+  private static final String AURORA_POSTGRES_USERNAME = "pgadmin";
+  private static final String AURORA_POSTGRES_PASSWORD = "my_password_2020";
+
+
+  protected static final String QUERY_1 = "select " +
+      "l_returnflag, " +
+      "l_linestatus, " +
+      "sum(l_quantity) as sum_qty, " +
+      "sum(l_extendedprice) as sum_base_price, " +
+      "sum(l_extendedprice * (1 - l_discount)) as sum_disc_price, " +
+      "sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge, " +
+      "avg(l_quantity) as avg_qty, " +
+      "avg(l_extendedprice) as avg_price, " +
+      "avg(l_discount) as avg_disc, " +
+      "count(*) as count_order " +
+      "from " +
+      "LINEITEM " +
+      "where " +
+      "l_shipdate <= date '1998-12-01' - interval '110' day " +
+      "group by " +
+      "l_returnflag, " +
+      "l_linestatus " +
+      "order by " +
+      "l_returnflag, " +
+      "l_linestatus;";
+
+  public static void setUp() throws SQLException {
+    if (!org.postgresql.Driver.isRegistered()) {
+      org.postgresql.Driver.register();
+    }
+
+    if (!Driver.isRegistered()) {
```

Comment:
why is this necessary ?

Reply:
We generate the META-INF file during compilation, so we use this to register the driver in the classpath in tests.

Reply:
So you are saying that the META-INF service file isn't available at this point ?

Reply:
> So you are saying that the META-INF service file isn't available at this point ?

No